esapi/esapi_integration_test.go

```diff
@@ -0,0 +1,39 @@
+// +build integration
+
+package esapi_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/elastic/go-elasticsearch"
+)
+
+func TestAPI(t *testing.T) {
+	t.Run("Search", func(t *testing.T) {
+		es, err := elasticsearch.NewDefaultClient()
+		if err != nil {
+			log.Fatalf("Error creating the client: %s\n", err)
+		}
+
+		res, err := es.Search(es.Search.WithTimeout(500 * time.Millisecond))
+		if err != nil {
+			log.Fatalf("Error getting the response: %s\n", err)
+		}
+		defer res.Body.Close()
+
+		if res.IsError() {
+			t.Fatalf("Error response: %s", res.String())
+		}
+
+		var d map[string]interface{}
+		err = json.NewDecoder(res.Body).Decode(&d)
+		if err != nil {
+			log.Fatalf("Error parsing the response: %s\n", err)
+		}
+		fmt.Printf("took=%vms\n", d["took"])
```

Comment:
For logging in test cases I think it’s most common to use the `t.Log` and `t.Logf` functions. The logging is enabled when the `-v` flag is enabled or if the test case fails.

Reply:
Here the `fmt` use is intentional. I use https://github.com/kyoh86/richgo as an enhanced runner, and it displays content printed by `fmt` and friends as "faint". The `t.Log/f` output is displayed prominently, so I usually not use it, since it interferes with the test error output...